subprojects/core-api/src/main/java/org/gradle/process/JavaForkOptions.java

```diff
@@ -220,11 +223,25 @@
     /**
      * Enable or disable debugging for the process. When enabled, the process is started suspended and listening on port
      * 5005.
+     * <p>
+     * Since Gradle 5.6, you can configure the port and other Java debug properties via
+     * {@link #setDebugOptions(JavaDebugOptions)}.
      *
      * @param enabled true to enable debugging, false to disable.
      */
     void setDebug(boolean enabled);
 
+    /**
+     * Configures Java Debug Wire Protocol properties for the process. If {@link #setDebug(boolean)} is enabled then
+     * the {@code -agentlib:jdwp=...}  will be appended to the JVM arguments with the configuration from the parameter.
+     *
+     * @param debugOptions the Java debug configuration
+     * @since 5.6
+     */
+    @Input
+    @Incubating
+    void setDebugOptions(JavaDebugOptions debugOptions);
```

Comment:
The `@Input` here doesn't work (it's on a setter), but I think we should replace this with...

1. Replace `@Input` to `@Internal` on `getDebug` (assuming we add enable to JavaDebugOptions)
2. Add a `@Nested JavaDebugOptions getDebugOptions()`. This will make us look at the JavaDebugOptions for more input properties.
3. Add a `void debugOptions(Action<JavaDebugOptions>)`

Instead of explicitly setting a new `JavaDebugOptions`, a user would do...
```
javaexec {
   debugOptions {
       enabled = true
       suspend = false
       port = 12345
   }
}
```